src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useState } from 'react';
+import { Loader } from '../Loader';
+import { Person } from '../../types';
+import { getPeople } from '../../api';
+import { PeopleTable } from '../PeopleTable';
+import { PeopleContext } from '../../peopleContext';
+import { PeopleFilters } from '../PeopleFilters';
+import { useSearchParams } from 'react-router-dom';
+import { filterPeople } from '../../utils/filterPeople';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [filteredPeople, setFilteredPeople] = useState<Person[]>(people);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const [searchParams] = useSearchParams();
```

Comment:
`useSearchParams` returns a `URLSearchParams` instance which is live and updates as the search string of the current URL changes. This means you don't need to include `searchParams` as a dependency in your useEffect hook, because it will always refer to the current search parameters.